redis_cluster_handler.go

```diff
@@ -128,13 +128,25 @@ func (r *RedisClusterStorageManager) cleanKey(keyName string) string {
 	return setKeyName
 }
 
-// GetKey will retrieve a key from the database
-func (r *RedisClusterStorageManager) GetKey(keyName string) (string, error) {
-	if GetRelevantClusterReference(r.IsCache) == nil {
-		log.Info("Connection dropped, connecting..")
+func (r *RedisClusterStorageManager) ensureConnection() {
+	if GetRelevantClusterReference(r.IsCache) != nil {
+		// already connected
+		return
+	}
+	log.Info("Connection dropped, reconnecting...")
+	for {
```

Comment:
Why is it inside `for` loop?
`r.Connect` can never return `false` and `GetRelevantClusterReference` seems like never can be `nil` if `Connect` was called.